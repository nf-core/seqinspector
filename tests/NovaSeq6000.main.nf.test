nextflow_pipeline {

    name "Test Workflow main.nf on NovaSeq6000 data"
    script "../main.nf"
    tag "seqinspector"
    tag "PIPELINE"

    test("NovaSeq6000 data test") {

        when {
            config "./NovaSeq6000.main.nf.test.config"
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/group_reports/lane1/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/group_reports/lane1/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/group_reports/lane1/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/group_reports/lane1/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/group_reports/group1/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/group_reports/group1/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/group_reports/group1/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/group_reports/group1/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/group_reports/group2/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/group_reports/group2/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/group_reports/group2/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/group_reports/group2/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/group_reports/test/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/group_reports/test/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/group_reports/test/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/group_reports/test/multiqc_data/multiqc_software_versions.txt"),
                    ).match()
                },
            )
        }
    }
}
