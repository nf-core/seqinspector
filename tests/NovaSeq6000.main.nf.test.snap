{
    "NovaSeq6000 data test": {
        "content": [
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,3730f9046b20ac5c17a86db0a33f8d5d",
            "multiqc_general_stats.txt:md5,d521de54d1e659bf7892105f7d23d4db",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,8284e25ccc21041cf3b5a32eb6a51e78",
            "multiqc_general_stats.txt:md5,d52544eb1a505c889a2f9117cf94a5fa",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,f38ffdc112c73af3a41ed15848a3761f",
            "multiqc_general_stats.txt:md5,5b1190093085ef073d4bd5818c9cde79",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,7ff71ceb8ecdf086331047f8860c3347",
            "multiqc_general_stats.txt:md5,79c1090dd8a97912893f8491641b9dc9",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,519ff344a896ac369bba4d5c5b8be7b5",
            "multiqc_general_stats.txt:md5,41611bd5ab9e79425c466bf976b03bdc",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T08:58:26.732622"
    }
}